discovery/openstack/openstack.go

```diff
@@ -96,12 +97,21 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
+
+	if c.EndpointType != "" &&
```

Comment:
You can set `EndpointType` to `public` in `DefaultSDConfig` and remove this check.